src/main/java/org/embulk/output/snowflake/SnowflakeCopyBatchInsert.java

```diff
@@ -84,18 +87,14 @@ protected BufferedWriter openWriter(File newFile) throws IOException {
   }
 
   public int getBatchWeight() {
-    long fsize = currentFile.length();
-    if (fsize > Integer.MAX_VALUE) {
-      return Integer.MAX_VALUE;
-    } else {
-      return (int) fsize;
-    }
+    return batchWeight;
```

Comment:
The current implementation uses stream output file size, which returns the fixed value. It never reaches the threshold.